### ISSUE TITLE
Valid XML not passed validation

### ISSUE THREAD

**Issue by senleft** [jcabi/jcabi-xml#125]
Please see full stacktrace at http://pastebin.com/zK7HkMK8

Patterns.XML not matches specified XML (attached as planets.txt due github file extension restriction) as valid XML.
[planets.txt](https://github.com/jcabi/jcabi-xml/files/491642/planets.txt)

**Comment by dmarkov** [comment]
@yegor256 please do something about this issue

**Comment by dmarkov** [comment]
@amihaiemil it's yours, please help

**Comment by amihaiemil** [comment]
@senleft can you give me the ``planets.xsl``? I am running a test with planets.xml file you attached and I'm getting the following error:

![image](https://cloud.githubusercontent.com/assets/6305156/20515313/b10826b6-b099-11e6-8f3a-605bdc43e061.png)


Which makes me thing the whole xml passes patterns validation but it doesn't find the ``xsl`` and so it fails.

**Comment by amihaiemil** [comment]
@senleft I figured why I got the exception above it's not lack of xsl. 
I also did try to instantiate an XmlDocument with the exact contents of planets.xml you attached and it worked

![image](https://cloud.githubusercontent.com/assets/6305156/20515696/6d6a2af6-b09b-11e6-9331-c72834079215.png)

And that XmlDocument constructor calls DomParser:

![image](https://cloud.githubusercontent.com/assets/6305156/20515722/a965940a-b09b-11e6-85a0-7b11c4caee9f.png)

which runs that regex pattern. So as it is now, I'd say I did not reproduce it? 
Are you absolutely sure the xml was valid at the moment when you tried? :D

**Comment by senleft** [comment]
@amihaiemil Yes, I'm absolutely sure about it. I will try to reproduce it and attach the full project source.

**Comment by amihaiemil** [comment]
@senleft sure, thanks. I will wait for that since I'm out of leads for the moment :)

**Comment by amihaiemil** [comment]
@dmarkov assign someone else; no more time for this :(

**Comment by amihaiemil** [comment]
@senleft I reproduced the problem (it's indeed a regex problem), but got no more time for this. Someone else should take it up soon. In the meantine, please don't turn off notifications from this post as in the end, you should be the one to close it. Thanks!

**Comment by dmarkov** [comment]
> @dmarkov assign someone else; no more time for this :( 

@amihaiemil 30 points was deducted from your rating

**Comment by dmarkov** [comment]
> @dmarkov assign someone else; no more time for this :( 

@amihaiemil someone else will help in this task, no problem at all

**Comment by eglobetrotter** [comment]
I'm getting the same issue, here, I have absolutely valid XML and the pattern returns this exception:
java.lang.IllegalArgumentException: Doesn't look like XML: 'content of my valid xml file'

The issue seems to be in this string transformation: txt.replaceAll("\\s", "") before pattern check. For any reason it not only removes lines returns but also whitespaces, so from ﻿`<?xml version="1.0" encoding="utf-8"?>` it makes `﻿<?xmlversion="1.0"encoding="utf-8"?>`.

This is my Java Runtime:
openjdk version "1.8.0_191"
OpenJDK Runtime Environment (build 1.8.0_191-8u191-b12-0ubuntu0.18.04.1-b12)
OpenJDK 64-Bit Server VM (build 25.191-b12, mixed mode)
